specification/cognitiveservices/data-plane/ContentSafety/preview/2023-11-30-preview/contentsafety.json

```diff
@@ -0,0 +1,3102 @@
+      ]
+    },
+    "TextIncidentCreateOrUpdate": {
+      "type": "object",
+      "description": "Text Incident.",
+      "properties": {
+        "incidentName": {
+          "type": "string",
+          "description": "incident name.",
+          "maxLength": 64,
+          "pattern": "^[0-9A-Za-z._~-]+$",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "description": {
+          "type": "string",
+          "description": "Incident description.",
+          "maxLength": 1024
+        }
+      }
+    },
+    "TextIncidentSample": {
+      "type": "object",
+      "description": "Sample in a Text Incident.",
+      "properties": {
+        "incidentSampleId": {
+          "type": "string",
+          "description": "incident name.",
+          "maxLength": 64,
+          "readOnly": true
+        },
+        "text": {
+          "type": "string",
+          "description": "IncidentSample text content.",
+          "maxLength": 1024
+        }
+      },
+      "required": [
+        "incidentSampleId"
+      ]
+    },
+    "TextTask": {
+      "type": "string",
+      "description": "Text analyze task.",
+      "enum": [
+        "QNA",
+        "Chat",
+        "Summarization"
+      ],
+      "x-ms-enum": {
+        "name": "TextTask",
+        "modelAsString": true
+      }
+    },
+    "UngroundedTextDetails": {
+      "type": "object",
+      "description": "The detailed information about a text identified as ungroundness.",
+      "properties": {
+        "ungroundedTextOffset": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The index of the ungrounded text from the beginning of the original text."
+        },
+        "ungroundedTextLength": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The length of the ungrounded text."
+        },
+        "ungroundedText": {
+          "type": "string",
+          "description": "The ungrounded text."
+        },
+        "ungroundedTextReason": {
+          "type": "string",
+          "description": "The reason or explanation for identifying the text as ungrounded."
+        }
+      },
+      "required": [
+        "ungroundedTextOffset",
+        "ungroundedTextLength",
+        "ungroundedText",
+        "ungroundedTextReason"
+      ]
+    },
+    "UngroundednessAnalysis": {
+      "type": "object",
+      "description": "The detection result of ungroudedness.",
+      "properties": {
+        "isUngrounded": {
+          "type": "boolean",
+          "description": "Detection result for ungrouded text."
+        },
+        "confidenceScore": {
+          "type": "number",
+          "format": "float",
+          "description": "Model confidence store in the analysis results."
+        },
+        "ungroundednessPercentage": {
+          "type": "number",
+          "format": "float",
+          "description": "Percentage of Ungrounded Text."
```

Comment:
We may indicate whether this is based on the number of sentence or number of characters?